Role permissions order tool and workflow

peternied opened the pull request:
### Description
Adds a NodeJS tool that can inspect yaml role definitions, check if they are in alphabetical order, correct them if required.  Seems pretty useful to prevent manual inspection of changes to the roles definitions, but curious what other folks think.

### Issues Resolved
- Related #2732

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

DarshitChanpura reviewed (dismissed):
Great addition @peternied !

davidlago commented on check-permissions-order.js:
```diff
@@ -0,0 +1,75 @@
+
+    const indexPermissions = role?.value?.items?.filter(item => item.key && item.key.value === 'index_permissions')
+    const allowedIndexActions = indexPermissions?.[0]?.value?.items[0]?.items?.filter(item => item.key && item.key.value === 'allowed_actions')
+    if (!allowedIndexActions) { return }
```
Nit: lines 22-32 and 36-46 can be DRY-ed up

davidlago reviewed (approved):
Is this something we were previously documenting and/or enforcing via PR review? Thanks for taking the humans out of the equation @peternied :)

RyanL1997 commented:
Could we rebase and re-run the CIs? It is kinda out of date for 34commits behind.

scrawfor99 commented:
This just needs to have the new check rerun on it after the rebase and then it can be approved and merged.

peternied commented:
Thanks for the reviews @RyanL1997 @DarshitChanpura @davidlago @scrawfor99 

I've cleaned up the JavaScript code and reverted the roles files changed and then re-ran the tool to avoid any merge conflicts

willyborankin commented:
Tests failures are relevant. e.g:
`RoleBasedAccessTests` L174

```java
        Assert.assertEquals(
            new SecurityJsonNode(DefaultObjectMapper.readTree(response.getBody())).getDotted(
                "opendistro_security_role_starfleet_captains.cluster_permissions"
            ).get(0).asString(),
            "cluster:monitor*"
        );

````
expects  get `cluster_permissions`  at the first position in the index since it was in this position before sorting 

before sorting:
```yaml
    - "cluster:monitor*"
    - "*bulk*"
```
after sorting:
```yaml
    - "*bulk*"
    - "cluster:monitor*"
```

peternied commented on src/test/java/org/opensearch/security/dlic/rest/api/RoleBasedAccessTest.java:
```diff
@@ -173,11 +177,9 @@ public void testActionGroupsApi() throws Exception {
-                "opendistro_security_role_starfleet_captains.cluster_permissions"
-            ).get(0).asString(),
-            "cluster:monitor*"
+        assertThat(
```
Props to @willyborankin who figured this out so quickly I didn't even get a chance to look into it.

Upgraded this test to make sure both permissions are present instead of only the first one.
